src/AutoRest.CSharp/Mgmt/Models/MgmtRestOperation.cs

```diff
@@ -385,14 +385,17 @@ private CSharpType GetWrappedMgmtReturnType(CSharpType? originalType)
             if (!IsResourceDataType(originalType, out var data))
                 return originalType;
 
-            if (Resource is not null && Resource.ResourceData.Type.Equals(originalType))
+            bool shouldReturnData = Configuration.MgmtConfiguration.DisableResourceReturn.Contains(OperationId);
+            if (Resource is not null && Resource.ResourceData.Type.Equals(originalType) && !shouldReturnData)
                 return Resource.Type;
 
             var foundResources = MgmtContext.Library.ArmResources.Where(resource => resource.ResourceData.Type.Equals(originalType)).ToList();
             return foundResources.Count switch
             {
                 0 => throw new InvalidOperationException($"No resource corresponding to {originalType?.Name} is found"),
-                1 => foundResources.Single().Type,
+                1 => shouldReturnData ?
+                    throw new InvalidOperationException($"When resource data is only used by one resource, we don't support directly returning the data.") :
+                    foundResources.Single().Type,
```

Comment:
What is the usage if this configuration? I only see it is used here, and if it is configured, we will throw an exception?

Reply:
When I fix the issue in GuestConfiguration, I noticed that when the data model is the same as the data in the resource, this [check](https://github.com/Azure/autorest.csharp/blob/10b43d4cc1bc01f5c114501fe89a050683b69d1d/src/AutoRest.CSharp/Mgmt/Models/MgmtRestOperation.cs#L389) will be true, which means that we will return the resource even though the data model is shared. And I believe this is not expected behavior.